src/components/timeline/TrackCounterGraph.js

```diff
@@ -121,12 +139,24 @@ class TrackMemoryCanvas extends React.PureComponent<CanvasProps> {
           // This is the first iteration, only move the line.
           ctx.moveTo(x, y);
         } else {
+          if (graphType === GRAPH_TYPE.visualProgress) {
+            // Plotting a extra point to generate a step-wise graph for visual progress.
+            ctx.lineTo(x, yOld);
+          }
           ctx.lineTo(x, y);
         }
+        yOld = y;
+      }
+      if (graphType === GRAPH_TYPE.visualProgress) {
```

Comment:
I'm not quite sure what the best way to make this a generic feature rather than a graph type check. I think maybe we should just rely on the original behavior. Is there a reason not to? Also I think this implementation has an issue in that it will draw into the empty thread indicators.